src/System.Private.DeveloperExperience.Console/src/Internal/DeveloperExperience/ConsolePal.Windows.cs

```diff
@@ -3,32 +3,40 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Text;
 
 namespace Internal.DeveloperExperience
 {
     internal static class ConsolePal
     {
-        private static IntPtr s_consoleErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
+        private static IntPtr s_stdErrorHandle = Interop.mincore.GetStdHandle(Interop.mincore.HandleTypes.STD_ERROR_HANDLE);
 
         internal unsafe static void WriteError(string errorMessage)
         {
-            if (s_consoleErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
+            if (s_stdErrorHandle == new IntPtr(Interop.mincore.HandleTypes.INVALID_HANDLE_VALUE))
             {
                 // ensure we have a valid handle before writing to it
                 return; 
             }
 
-            fixed (char *pBuffer = errorMessage)
+            // Ensure that the encoding we use matches the current code page.
+            int currentCodePage = (int)Interop.mincore.GetConsoleOutputCP();
+            Encoding encoding;
+            try
             {
-                int numberOfCharsWritten;
-                Interop.mincore.WriteConsole(s_consoleErrorHandle, (byte*)pBuffer, errorMessage.Length, out numberOfCharsWritten, IntPtr.Zero);
+                encoding = Encoding.GetEncoding(currentCodePage);
```

Comment:
System.Diagnostics.Debug declares in the following project.json. I don't know if there is real dependency on this specific System.Private.DeveloperExperience.Console
@weshaggard @AtsushiKan please advise if it is k to have System.Private.DeveloperExperience.Console take dependency on System.Console


    "netcore50": {
      "dependencies": {
        "Microsoft.TargetingPack.Private.NETNative": "1.1.0-beta-24718-00"
      }
    },


and this will be expanded to 

    ".NETCore,Version=v5.0": {
      "Microsoft.TargetingPack.Private.NETNative/1.1.0-beta-24718-00": {
        "type": "package",
        "compile": {
          "lib/netcore50/System.Private.CompatQuirks.dll": {},
          "lib/netcore50/System.Private.CoreLib.Augments.dll": {},
          "lib/netcore50/System.Private.CoreLib.DynamicDelegate.dll": {},
          "lib/netcore50/System.Private.CoreLib.InteropServices.dll": {},
          "lib/netcore50/System.Private.CoreLib.Threading.dll": {},
          "lib/netcore50/System.Private.CoreLib.WinRTInterop.dll": {},
          "lib/netcore50/System.Private.CoreLib.dll": {},
          "lib/netcore50/System.Private.DataContractSerialization.dll": {},
          "lib/netcore50/System.Private.DeveloperExperience.AppX.dll": {},
          "lib/netcore50/System.Private.DeveloperExperience.Console.dll": {}, <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
          "lib/netcore50/System.Private.Interop.Extensions.dll": {},
          "lib/netcore50/System.Private.Interop.dll": {},
          "lib/netcore50/System.Private.PortableServiceModelThunks.dll": {},
          "lib/netcore50/System.Private.PortableThunks.dll": {},
          "lib/netcore50/System.Private.Reflection.Core.dll": {},
          "lib/netcore50/System.Private.Reflection.Execution.dll": {},
          "lib/netcore50/System.Private.Reflection.Extensibility.dll": {},
          "lib/netcore50/System.Private.Reflection.Metadata.dll": {},
          "lib/netcore50/System.Private.Reflection.MissingMetadataException.dll": {},
          "lib/netcore50/System.Private.Reflection.dll": {},
          "lib/netcore50/System.Private.ServiceModel.dll": {},
          "lib/netcore50/System.Private.StackTraceGenerator.dll": {},
          "lib/netcore50/System.Private.Threading.AsyncCausalitySupport.dll": {},
          "lib/netcore50/System.Private.Threading.dll": {},
          "lib/netcore50/System.Private.TypeLoader.dll": {},
          "lib/netcore50/System.Private.WinRTInterop.CoreLib.dll": {}


---
In reply to: [88763016](https://github.com/dotnet/corert/pull/2235#discussion_r88763016) [](ancestors = 88763016)